Sources/Build/BuildDelegate.swift

```diff
@@ -0,0 +1,223 @@
+    
+    /// Called to allow the delegate to skip commands without cancelling their
+    /// dependents. See llbuild's should_command_start.
+    public func shouldCommandStart(_ command: llbuildSwift.Command) -> Bool {
+        return true
+    }
+
+    /// Called when a command has been finished.
+    public func commandFinished(_ command: llbuildSwift.Command, result: CommandResult) {
+    }
+
+    /// Called to report an error during the execution of a command.
+    public func commandHadError(_ command: llbuildSwift.Command, message: String) {
+        diagnostics.emit(data: LLBuildCommandErrorDiagnostic(message: message))
+    }
+
+    /// Called to report a note during the execution of a command.
+    public func commandHadNote(_ command: llbuildSwift.Command, message: String) {
+        outputStream <<< message <<< "\n"
+    }
+
+    /// Called to report a warning during the execution of a command.
+    public func commandHadWarning(_ command: llbuildSwift.Command, message: String) {
+        diagnostics.emit(data: LLBuildCommandWarningDiagnostic(message: message))
+    }
+
+    /// Called by the build system to report a command could not build due to
+    /// missing inputs.
+    public func commandCannotBuildOutputDueToMissingInputs(
+        _ command: llbuildSwift.Command,
+        output: BuildKey,
+        inputs: [BuildKey]
+    ) {
+    }
+
+    /// Called by the build system to report a node could not be built
+    /// because multiple commands are producing it.
+    public func cannotBuildNodeDueToMultipleProducers(output: BuildKey, commands: [llbuildSwift.Command]) {
+    }
+
+    /// Called when a command's job has started executing an external process.
+    ///
+    /// The system guarantees that any commandProcessStarted() call will be paired
+    /// with exactly one \see commandProcessFinished() call.
+    ///
+    /// - parameter process: A unique handle used in subsequent delegate calls
+    /// to identify the process. This handle should only be used to associate
+    /// different status calls relating to the same process. It is only
+    /// guaranteed to be unique from when it has been provided here to when it
+    /// has been provided to the \see commandProcessFinished() call.
+    public func commandProcessStarted(_ command: llbuildSwift.Command, process: ProcessHandle) {
+    }
+
+    /// Called to report an error in the management of a command process.
+    ///
+    /// - parameter process: The process handle.
+    /// - parameter message: The error message.
+    public func commandProcessHadError(_ command: llbuildSwift.Command, process: ProcessHandle, message: String) {
+    }
+
+    /// Called to report a command processes' (merged) standard output and error.
+    ///
+    /// - parameter process: The process handle.
+    /// - parameter data: The process output.
+    public func commandProcessHadOutput(_ command: llbuildSwift.Command, process: ProcessHandle, data: [UInt8]) {
+        outputStream <<< data
```

Comment:
I wonder if we should be flushing here... I suspect so.

Reply:
Correct. Done.